### PR TITLE
Misc re-frame tweaks

### PR DESCRIPTION
**Pull request by danielcompton**
Not ready for merge, as much a test of reviewable.io as anything.

---
This change is [Reviewable](https://reviewable.io/reviews/day8/re-frame/205)

### CODE REVIEW

**Comment by danielcompton** [comment]
Review status: 0 of 7 files reviewed at latest revision, 3 unresolved discussions.

---

*[src/re_frame/fx.cljc, line 36 \[r2\]](https://reviewable.io:443/reviews/day8/re-frame/205#-KPrVfVKbArk05euAH00:-KPrVfVKbArk05euAH01:boioa8) ([raw file](https://github.com/day8/re-frame/blob/41e62cce85943942388c2721d9baac557312f36a/src/re_frame/fx.cljc#L36)):*
> ```Clojure
>     :after (fn do-fx-after
>              [context]
>              (doseq [[effect-k value] (:effects context)]
> ```

I don't think we do anything with the values returned by these effects? Is that correct?

---

*[src/re_frame/fx.cljc, line 107 \[r2\]](https://reviewable.io:443/reviews/day8/re-frame/205#-KPrVtqJPr7otgIp1g4H:-KPrVtqJPr7otgIp1g4I:b-v4b612) ([raw file](https://github.com/day8/re-frame/blob/41e62cce85943942388c2721d9baac557312f36a/src/re_frame/fx.cljc#L107)):*
> ```Clojure
>     (let [clear-event (partial clear-handlers events/kind)]
>       (if (sequential? value)
>         (doseq [event (if (sequential? value) value [value])]
> ```

Prefer doseq over `(doall (map`. https://stuartsierra.com/2015/08/25/clojure-donts-lazy-effects

---

*[src/re_frame/subs.cljc, line 90 \[r2\]](https://reviewable.io:443/reviews/day8/re-frame/205#-KPrW70SofKWvJHvbfVU:-KPrW70SofKWvJHvbfVV:b-j64z38) ([raw file](https://github.com/day8/re-frame/blob/41e62cce85943942388c2721d9baac557312f36a/src/re_frame/subs.cljc#L90)):*
> ```Clojure
>   (map-vals inc {:a 4, :b 2}) => {:a 5, :b 3}"
>   [f m]
>   (into (empty m)
> ```

Transducer

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/day8/re-frame/205)*

**Comment by danielcompton** [comment]
Review status: 0 of 7 files reviewed at latest revision, 3 unresolved discussions.

---

*[src/re_frame/std_interceptors.cljc, line 66 \[r3\]](https://reviewable.io:443/reviews/day8/re-frame/205#-KPrYENZmfkeWNFvx7Ut:-KPrYENZmfkeWNFvx7Uu:b7ho4tr) ([raw file](https://github.com/day8/re-frame/blob/569277c7d0556fd9986ff109e314291b91fd6dce/src/re_frame/std_interceptors.cljc#L66)):*
> ```Clojure
>     :before  (fn trimv-before
>                [context]
>                (update context cofx/kind subvec 1))))
> ```

Am I a bad re-framer for updating context directly like this?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/day8/re-frame/205)*

**Comment by mike-thompson-day8** [comment]
Changes manually merged.  Thanks!

File: src/re_frame/registrar.cljc [hunk 1]
```diff
@@ -26,15 +26,15 @@
 
   ([kind id required?]
    (let [handler (get-handler kind id)]
-     (when debug-enabled?
-       (when (and required? (nil? handler))
+     (when debug-enabled?                                   ;; This is in a separate when so Closure DCE can run
```
Comment (lewang) [inline src/re_frame/registrar.cljc]: @danielcompton What does comment mean?  What is DCE?

Comment (martinklepsch) [inline src/re_frame/registrar.cljc]: DCE is an abbreviation for *Dead Code Elimination*, a feature of the Google Closure compiler.

**Pull request merged**